### ISSUE TITLE
Add complex autograd support for torch.linalg.det

### ISSUE THREAD

**Issue by anjali411** [pytorch/pytorch#53513]
```
>>> a = torch.randn(2, 2, dtype=torch.cdouble, requires_grad=True)
>>> torch.linalg.det(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: det does not support automatic differentiation for outputs with complex dtype.
```

cc @ezyang @anjali411 @dylanbespalko @mruberry @aocsa

**Comment by mruberry** [comment]
Let's make this issue for torch.linalg.det(). torch.det() is already docs-deprecated.

**Comment by kurtamohler** [comment]
Just FYI, an OpInfo for `linalg.det` is being added in #53119, so when I submit a PR for this, I think it might be best to add a standalone test for complex autograd in `test_linalg.py` for the time being. Then once #53119 is merged, I can update my PR or make a followup PR to enable complex autograd for the OpInfo and remove the temporary test.

**Comment by kurtamohler** [comment]
There's a dependency on `nonzero`, which does not have complex support yet. I may need to wait until #50655 is finished. However, it might be possible to rewrite to avoid `nonzero` and be a little bit faster. I'll look into it

**Comment by nikitaved** [comment]
Closing, this was fixed in https://github.com/pytorch/pytorch/pull/61905.
